openmls/src/group/core_group/proposals.rs

```diff
@@ -150,7 +149,7 @@ impl ProposalQueue {
                 ProposalOrRef::Proposal(proposal) => {
                     // ValSem200
                     if let Proposal::Remove(ref remove_proposal) = proposal {
-                        if remove_proposal.removed() == sender.sender {
+                        if remove_proposal.removed() == sender.sender && sender.is_member() {
```

Comment:
Could you add a comment here why the sender type has to be `member`?

Reply:
Should that not go into the description of `ValSem200` instead?

Reply:
My question was: why are you introducing this restriction?

Reply:
Ah, the point is that as long as we're using leaf indices, the sender of the proposal might be the same as the committer. In case of the external commit, this is exactly what we want, so we have to allow it. This should go away with KeyPackage hashes as `sender` identifiers, though.